utils/visualizer/qviz/content_loader.py

```diff
@@ -2,250 +2,261 @@
 import os
 import glob
 from typing import Optional
-
+import pandas as pd
+from deltalake import DeltaTable
 import pyarrow.parquet as pa
+from collections import defaultdict
+from qviz.cube import Cube, SamplingInfo, normalize_weight
 
 
-def process_table_delta_log(table_path: str, revision_id: str) -> (list[dict], Optional[dict]):
+# Returns metadata (dict), cubes(dict), root(cube), nodes & edges (list[dict])
+def process_table(
+    table_path: str, revision_id: str
+) -> tuple[dict, dict, Cube, list[dict]]:
     """
-    Extract valid AddFiles from table's _delta_log
-    :param table_path: path to qbeast table
-    :param revision_id: target RevisionID
-    :return: a list of AddFile dictionaries and optionally the metadata dictionary
+    Process a delta table given a path to a Qbeast table and a revision ID.
+    Example json log file content:
+        {"commitInfo":
+            {"timestamp":1726134717979,
+            "operation":"WRITE",
+            "operationParameters":
+                {"mode":"Overwrite"},
+                "isolationLevel":"Serializable",
+                "isBlindAppend":false,
+                "operationMetrics":{"numFiles":"16",
+                "numOutputRows":"300000",
+                "numOutputBytes":"9486957"},
+            "engineInfo":"Apache-Spark/3.5.0 Delta-Lake/3.1.0",
+            "txnId":"450ebb6f-6845-4e75-a1b5-d1143a1b9f29"}}
+        {"metaData":
+            {"id":"d8ed5bdc-9625-4cf2-940e-3ffe0a36104b",
+            "format":
+                {"provider":"parquet",
+                "options":{}},
+                "schemaString":
+                    "{\"type\":\"struct\",\"fields\":[
+                        {\"name\":\"event_time\",\"type\":\"timestamp\",\"nullable\":true,\"metadata\":{}},
+                        {\"name\":\"event_type\",\"type\":\"string\",\"nullable\":true,\"metadata\":{}},
+                        ...}}
+        {"protocol":
+            {"minReaderVersion":1,
+            "minWriterVersion":2}}
+        {"add":
+            {"path":"8eb0fcd7-c5ca-4b4c-a967-bc10188be935.parquet",
+            "partitionValues":{},
+            "size":814021,
+            "modificationTime":1726134717043,
+            "dataChange":true,
+            "stats":"{
```

Comment:
`stats` and `partitionValues` can surely be removed.
Also, you don't need to keep too many blocks there.

Reply:
I kept stats, because there we have the tags which store the blocks of the cubes